templates/auth_ui_settings.html

```diff
@@ -12,13 +12,201 @@
+    <p class="title primary-txt">
+      {{ template "settings-page-secondary-totp-title" }}
+    </p>
+    {{ $totp := false }}
+
+    {{ range $.Authenticators }}
+    {{ if eq .Type "totp" }}{{ if eq .Kind "secondary" }}
+    {{ $totp = true }}
+    {{ end }}{{ end }}
+    {{ end }}
+    {{ if $totp }}
+    <p class="description good-txt">
+      {{ template "activated-label" }}
+    </p>
+    {{ else }}
+    <p class="description warn-txt">
+      {{ template "inactive-label" }}
+    </p>
+    {{ end }}
+
+    <a class="action" href="#">
+      {{ template "details-button-label" }}
+    </a>
+  </section>
+  {{ end }}
+
+  <!-- Secondary OOB OTP -->
+  {{ if $.SecondaryOOBOTPEnabled }}
+  <section class="settings-row settings-page-section-with-title-desc-action">
+    <p class="title primary-txt">
+      {{ template "settings-page-secondary-oob-otp-title" }}
+    </p>
+    {{ $totp := false }}
+
+    {{ range $.Authenticators }}
+    {{ if eq .Type "oob_otp" }}{{ if eq .Kind "secondary" }}
+    {{ $totp = true }}
+    {{ end }}{{ end }}
+    {{ end }}
+    {{ if $totp }}
+    <p class="description good-txt">
+      {{ template "activated-label" }}
+    </p>
+    {{ else }}
+    <p class="description warn-txt">
+      {{ template "inactive-label" }}
+    </p>
+    {{ end }}
+
+    <a class="action" href="#">
+      {{ template "details-button-label" }}
+    </a>
+  </section>
+  {{ end }}
+
+  <!-- Secondary Password -->
+  {{ if $.SecondaryPasswordEnabled }}
+  {{ $secondary_password := false }}
+
+  <!-- The user at most has 1 secondary password. -->
+  {{ range $.Authenticators }}
+  {{ if eq .Type "oob_otp" }}{{ if eq .Kind "secondary" }}
+  {{ $secondary_password = . }}
+  {{ end }}{{ end }}
+  {{ end }}
+
+  <section class="settings-row settings-page-section-with-title-desc-action">
+    <p class="title primary-txt">
+      {{ template "settings-page-secondary-password-title" }}
+    </p>
+
+    {{ if $secondary_password }}
+    <p class="description secondary-txt">
+      {{ template "settings-page-secondary-password-description" (makemap "time" "FIXME") }}
+    </p>
+    {{ end }}
+
+    {{ if $secondary_password }}
+    <a class="action" href="#">
+      {{ template "change-password-button-label" }}
+    </a>
+    {{ else }}
+    <a class="action" href="#">
+      {{ template "setup-button-label" }}
+    </a>
+    {{ end }}
+  </section>
+
+  {{ end }}
+
+  <!-- Recovery code -->
+  {{ if $.RecoveryCodes }}
+  <section class="settings-row settings-page-section-with-title-desc-action">
+    <p class="title primary-txt">
+      {{ template "settings-page-recovery-code-title" }}
+    </p>
+    <p class="description secondary-txt">
+      {{ template "settings-page-recovery-code-description" }}
+    </p>
+    <a class="action" href="#">
+      {{ template "show-recovery-code-button-label" }}
+    </a>
+  </section>
+  {{ end }}
+
+  <!-- Revoke all device tokens -->
+  {{ if $.RecoveryCodes }}
```

Comment:
It seems we're assuming 'have recovery codes' = 'MFA enabled'. I suppose we should lift the 'MFA enabled' flag to a field in view model to be more explicit.